mrmustard/physics/representations.py

```diff
@@ -248,8 +266,27 @@ def trace(self, idx_z: tuple[int, ...], idx_zconj: tuple[int, ...]) -> Bargmann:
         return self.__class__(math.astensor(A), math.astensor(b), math.astensor(c))
 
     def reorder(self, order: tuple[int, ...] | list[int]) -> Bargmann:
-        r"""Reorders the indices of the A matrix and b vector of an (A,b,c) triple.
-        Returns a new Bargmann object."""
+        r"""
+        Reorders the indices of the ``A`` matrix and ``b`` vector of the ``(A, b, c)`` triple in
+        this Bargmann object.
+
+        .. code-block::
+
+            >>> from mrmustard.physics.representations import Bargmann
+            >>> from mrmustard.physics.triples import displacement_gate_Abc
+
+            >>> rep_dgate1 = Bargmann(*displacement_gate_Abc([0.1, 0.2, 0.3]))
+            >>> rep_dgate2 = Bargmann(*displacement_gate_Abc([0.2, 0.3, 0.1]))
+
+            >>> assert rep_dgate1.reorder([1, 2, 0, 4, 5, 3]) == rep_dgate2
+
+        Args:
+            order: The new order.
+
+        Returns:
+            The reordered Bargmann object.
+        """
+        # @filippo, should we add error-checking? What if I pass an order that is incorrect (e.g., shorter)?
```

Comment:
@ziofil

Reply:
I think if the size is wrong something will break eventually, so it's not strictly needed. If we go this way we could consider allowing a shorter order to "extract" only those rows/cols on top of reordering (which is already happening on the A/b part via `gather`, but then `transpose` would complain if we have an array version of c). Otherwise we can check the length and move on.